iroha_p2p/src/network.rs

```diff
@@ -2,48 +2,66 @@ use std::{
     collections::{HashMap, HashSet},
     fmt::{Debug, Formatter},
     io,
-    net::SocketAddr,
 };
 
 use async_stream::stream;
 use futures::Stream;
 use iroha_actor::{
     broker::{Broker, BrokerMessage},
-    Actor, Addr, Context, Handler,
+    Actor, Addr, Context, ContextHandler, Handler,
 };
 use iroha_crypto::PublicKey;
-use iroha_logger::{info, warn};
+#[allow(unused_imports)]
+use iroha_logger::{debug, info, warn};
 use parity_scale_codec::{Decode, Encode};
-use tokio::net::{TcpListener, TcpStream};
+use tokio::{
+    net::{TcpListener, TcpStream},
+    sync::{
+        oneshot,
+        oneshot::{Receiver, Sender},
+    },
+};
 use ursa::{encryption::symm::Encryptor, kex::KeyExchangeScheme};
 
 use crate::{
     peer::{Peer, PeerId},
     Error,
 };
 
+/// Represents a Peer actor and its connection ID
+#[derive(Clone, Debug)]
+pub struct Connection<T, K, E>(Addr<Peer<T, K, E>>, ConnectionId)
+where
+    T: Debug + Encode + Decode + BrokerMessage + Send + Sync + Clone + 'static,
+    K: KeyExchangeScheme + Send + 'static,
+    E: Encryptor + Send + 'static;
+
 /// Main network layer structure, that is holding connections, called [`Peer`]s.
 pub struct NetworkBase<T, K, E>
 where
-    T: Encode + Decode + BrokerMessage + Send + Sync + Clone + 'static,
+    T: Debug + Encode + Decode + BrokerMessage + Send + Sync + Clone + 'static,
     K: KeyExchangeScheme + Send + 'static,
     E: Encryptor + Send + 'static,
 {
-    /// Current peers in any state
-    pub peers: HashMap<PeerId, Addr<Peer<T, K, E>>>,
-    /// A set of connected peers, ready to work with
-    pub connected_peers: HashSet<PublicKey>,
+    /// Listening to this address for incoming connections. Must parse into [`SocketAddr`].
+    listen_addr: String,
```

Comment:
If it must parse into `SocketAddr` why not make it `SocketAddr` from the start?
```suggestion
    listen_addr: SocketAddr,
```

Reply:
It is used in logging, better it to not to change, if for example we will have domain name or like in docker tests `iroha:1337`.

Reply:
Please update the docs then. It means that it will not always `Must parse into [`SocketAddr`]`